app/classes/query.rb

```diff
@@ -304,6 +304,14 @@ def self.lookup(*)
     Query::Base.lookup(*)
   end
 
+  def self.current_or_related_query(*)
```

Comment:
Why is Query a module and not a class?  The documentation at the top of this page goes even further and claims that it's a "model".  This may be outside the remit of this PR, but I'd like to understand what should be fixed here.  At minimum it seems like we should update the documentation so it's accurate.  I don't see an obvious reason why Query should not be a class, so maybe that's the right direction.

Reply:
Good question, this is still Jason's setup as far as I know. 

I agree it's a class, and furthermore I'm preparing to have it inherit from `ActiveModel` and adapt the validations. (That's what that little `Mappable:MinimalLocation` refactor last week was practice for. `Query` is pretty much the exact use case that the whole `ActiveModel::API` was designed around.)

So regardless, it may be changing soon. I'll try making it a class, but if it takes a lot of time i'm inclined to keep it this way until i can switch it to ActiveModel and attributes.

Reply:
Made it a class